controllers/store.js

```diff
@@ -1,34 +1,150 @@
+const Book = require("../models/book");
+const User = require("../models/user");
+const BookCopy = require("../models/bookCopy");
+
 var getAllBooks = (req, res) => {
     //TODO: access all books from the book model and render book list page
-    res.render("book_list", { books: [], title: "Books | Library" });
+    Book.find({}, (err, foundBooks) => {
+        if (err) {
+            console.log(err);
+        } else {
+            res.render("book_list", { books: foundBooks, title: "Books | Library" });
+        }
+    })
 }
 
 var getBook = (req, res) => {
     //TODO: access the book with a given id and render book detail page
+    const id = req.params.id;
+
+    Book.findById(id, (err, foundBook) => {
+        if (err) {
+            console.log(err);
+        } else {
+            // console.log(id, foundBook)
+            res.render("book_detail", { book: foundBook, title: "Book Details" });
+        }
+    })
 }
 
 var getLoanedBooks = (req, res) => {
-
     //TODO: access the books loaned for this user and render loaned books page
+    const user = req.user;
+
+    User.findById(user.id)
+        .populate({ path: "loaned_books", populate: { path: "book" }})
+        .then(copies => {
+            const books = copies.loaned_books;
+            res.render("loaned_books", { books: books, title: "Loaned Books" })
+        })
+        .catch(err => {
+            console.log(err);
+        });
 }
 
+
 var issueBook = (req, res) => {
-    
     // TODO: Extract necessary book details from request
     // return with appropriate status
     // Optionally redirect to page or display on same
+    const { user, body } = req;    
+
+    Book.findById(body.bid, (err, foundBook) => {
+        if (err) {
+            console.log(err);
+        } else {
+            var availableCopies = foundBook.available_copies;
+            console.log(availableCopies);
+            BookCopy.find({ book: body.bid }, (err, bookCopies) => {
+                if (err) {
+                    console.log(err);
+                } else {
+                    for(var bookCopy of bookCopies){
+                        if (availableCopies && bookCopy.status) {
+                            User.findByIdAndUpdate(user.id, { $addToSet: { loaned_books: [bookCopy.id] } }, (err, result) => {
+                                if (err) {
+                                    console.log(err);
+                                } else {
+                                    console.log("Successfully added book to the list of loaned books!");
+                                }
+                            });
+                            BookCopy.findByIdAndUpdate(bookCopy.id, { $set: { status: false, borrow_data: Date.now(), borrower: user.id } }, (err, result) => {
```

Comment:
Format `borrow_date` to make it more readable for the user.

Reply:
I tried that but i was having some issues with that and there were time constraints as well. I'll keep that in mind for next time around.